src/torchcodec/_core/custom_ops.cpp

```diff
@@ -381,6 +384,38 @@ OpsAudioFramesOutput get_frames_by_pts_in_range_audio(
   return makeOpsAudioFramesOutput(result);
 }
 
+at::Tensor wrapEncoderPointerToTensor(std::unique_ptr<Encoder> uniqueEncoder) {
+  Encoder* encoder = uniqueEncoder.release();
+
+  auto deleter = [encoder](void*) { delete encoder; };
+  at::Tensor tensor =
+      at::from_blob(encoder, {sizeof(Encoder)}, deleter, {at::kLong});
+  auto encoder_ = static_cast<Encoder*>(tensor.mutable_data_ptr());
+  TORCH_CHECK_EQ(encoder_, encoder) << "Encoder=" << encoder_;
+  return tensor;
+}
+
+Encoder* unwrapTensorToGetEncoder(at::Tensor& tensor) {
+  TORCH_INTERNAL_ASSERT(tensor.is_contiguous());
+  void* buffer = tensor.mutable_data_ptr();
+  Encoder* encoder = static_cast<Encoder*>(buffer);
+  return encoder;
+}
+
+at::Tensor create_encoder(
+    const at::Tensor wf,
+    int64_t sample_rate,
+    std::string_view file_name) {
+  std::unique_ptr<Encoder> uniqueEncoder =
+      std::make_unique<Encoder>(wf, static_cast<int>(sample_rate), file_name);
```

Comment:
We have to take in `sample_rate` as an `int64_t` because it's coming from Python and all Python ints become `int64_t`. And we have to turn it into an `int` because that's the type it is in the `AVFormatContext`, so that's all correct. But to be extra paranoid, we can do a `TORCH_CHECK(sample_rate <= std::numeric_limits<int>::max())`.